### PR TITLE
dbt kestra blog post

### PR DESCRIPTION
**Pull request by MartinRst**

### CODE REVIEW

File: content/blogs/2024-04-02-dbt-kestra.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
+---
+title: "Orchestrate dbt Jobs with Kestra"
+description: "delve into the usage we see among our users regarding dbt."
```
Comment (anna-geller) [inline content/blogs/2024-04-02-dbt-kestra.md:3]:
```suggestion
description: "Dive into the ways to use dbt in a, quite literally, transformative way!"
```

File: content/blogs/2024-04-02-dbt-kestra.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+author:
+  name: Martin-Pierre Roset
+  image: "mproset"
+image: /blogs/2024-04-02-dbt-kestra.jpg
+---
+
+
+At Kestra, we observe many users orchestrating their dbt workflows alongside various elements – data ingestion, alerting, machine learning training, and more. While SQL plays a crucial role, organizations need a solution to automate the entire pipeline. This is where Kestra's orchestration engine steps in.
+
+In this blog post, we will delve into the usage we see among our users regarding dbt.
```
Comment (anna-geller) [inline content/blogs/2024-04-02-dbt-kestra.md:15]:
```suggestion
At Kestra, we observe many users orchestrating their dbt workflows alongside various elements – data ingestion, alerting, machine learning training, and more. While SQL plays a crucial role, organizations need a solution to automate the entire pipeline. This is where Kestra's orchestration engine steps in.

In this blog post, we will delve into the usage of dbt among our users.
```

File: content/blogs/2024-04-02-dbt-kestra.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+
+## About dbt
+
+[dbt](https://www.getdbt.com/) is the most popular SQL framework in the data industry.
+
+**it’s a SQL-first transformation workflow that lets teams quickly and collaboratively deploy analytics code following software engineering best practices like modularity, portability, CI/CD, and documentation.**
+
+Pioneering the concept of a full-fledged SQL framework, dbt empowers users to construct intricate model dependencies and efficiently test their SQL queries using a straightforward YAML declaration syntax. This combination of functionality and ease of use has cemented dbt's position as a leader in data transformation.
+
+They offer two versions of their tools.  dbt Core, the open source command line tool and dbt Cloud, which is a subscription-based cloud version provides an additional interface layer. It handles hosting, as well as offering a managed Git repository and the ability to schedule dbt transformation jobs for various environments.
```
Comment (anna-geller) [inline content/blogs/2024-04-02-dbt-kestra.md:27]:
```suggestion
**it’s a SQL-first transformation workflow that lets teams quickly and collaboratively deploy analytics code following software engineering best practices like modularity, portability, CI/CD, and documentation.**

Pioneering the concept of a full-fledged SQL framework, dbt empowers users to construct intricate model dependencies and efficiently test their SQL queries using a straightforward YAML declaration syntax. This combination of functionality and ease of use has cemented dbt's position as a leader in data transformation.

They offer two versions of their tools. dbt Core, the open-source command line tool, and dbt Cloud, a subscription-based cloud version, provide an additional interface layer. The latter handles hosting, offers a managed Git repository, and can schedule dbt transformation jobs for various environments.
```

File: content/blogs/2024-04-02-dbt-kestra.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+
+---
+
+## Kestra Plugin for dbt
+
+You can leverage the [dbt plugin](https://kestra.io/plugins/plugin-dbt) to orchestrate dbt Cloud and dbt Core jobs. Ke After each scheduled or ad-hoc workflow execution, the Outputs tab in the Kestra UI allows you to download and preview all dbt build artifacts. The Gantt and Topology view additionally render the metadata to visualize dependencies and runtimes of your dbt models and tests. The dbt Cloud task provides convenient links to easily navigate between Kestra and dbt Cloud UI.
```
Comment (anna-geller) [inline content/blogs/2024-04-02-dbt-kestra.md:33]:
```suggestion
You can leverage the [dbt plugin](https://kestra.io/plugins/plugin-dbt) to orchestrate dbt Cloud and dbt Core jobs. After each scheduled or ad-hoc workflow execution, the Outputs tab in the Kestra UI allows you to download and preview all dbt build artifacts. The Gantt and Topology view additionally render the metadata to visualize dependencies and runtimes of your dbt models and tests. The dbt Cloud task provides convenient links to easily navigate Kestra and dbt Cloud UI.
```

File: content/blogs/2024-04-02-dbt-kestra.md [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+
+### Execute dbt CLI commands from Kestra
+
+The **`io.kestra.plugin.dbt.cli.DbtCLI`** can be used to orchestrate any dbt command. You can run dbt Core jobs and enhance the functionality of dbt Cloud tasks within the Kestra environment.
+
+The plugin offers various customizable properties to fine-tune dbt tasks. These include specifying dbt CLI commands (**`dbt deps`**, **`dbt build`**), choosing the execution environment (Docker for isolation or process-based execution), and setting up the environment before running dbt commands. This configuration ensures that dbt tasks are executed  with all dependencies correctly prepared.
+
+for example you can Launch a `dbt build` command on a  dbt project hosted on GitHub:
```
Comment (anna-geller) [inline content/blogs/2024-04-02-dbt-kestra.md:41]:
```suggestion
The **`io.kestra.plugin.dbt.cli.DbtCLI`** can be used to orchestrate any dbt command. You can run dbt Core jobs and enhance the functionality of dbt Cloud tasks within the Kestra environment.

The plugin offers various customizable properties to fine-tune dbt tasks. These include specifying dbt CLI commands (**`dbt deps`**, **`dbt build`**), choosing the execution environment (Docker for isolation or process-based execution), and setting up the environment before running dbt commands. This configuration ensures that dbt tasks are executed with all dependencies correctly prepared.

For example, you can Launch a `dbt build` command on a  dbt project hosted on GitHub:
```

File: content/blogs/2024-04-02-dbt-kestra.md [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+      commands:
+        - dbt build
+      profiles: |
+        my_dbt_project:
+          outputs:
+            dev:
+            type: duckdb
+            path: ":memory:"
+          target: dev
+
```
Comment (anna-geller) [inline content/blogs/2024-04-02-dbt-kestra.md:70]:
```suggestion
          target: dev
```

File: content/blogs/2024-04-02-dbt-kestra.md [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+```
+
+Or Install a custom dbt version and run `dbt deps` and `dbt build` commands:
```
Comment (anna-geller) [inline content/blogs/2024-04-02-dbt-kestra.md:73]:
```suggestion
You can also install a custom dbt version and run `dbt deps` and `dbt build` commands:
```

File: content/blogs/2024-04-02-dbt-kestra.md [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+                fixed_retries: 1
+                threads: 16
+                timeout_seconds: 300
+            target: dev
+
+```
+
+### Disable parsing
+
+The dbt tasks  have the option to turn off parsing dbt DAG with the `parseRunResults` flag, a boolean property allowing disabling parsing of the dbt manifest. If your dbt project is large, with hundreds or thousands of models and tests, parsing the manifest may be unnecessary. This flag allows you to turn off parsing the manifest and still get the results of the dbt job by inspecting the execution logs.
```
Comment (anna-geller) [inline content/blogs/2024-04-02-dbt-kestra.md:121]:
```suggestion
The dbt tasks have the option to turn off parsing dbt DAG with the `parseRunResults` flag, a boolean property allowing disabling parsing of the dbt manifest. If your dbt project is large, with hundreds or thousands of models and tests, parsing the manifest may be unnecessary. This flag allows you to turn off parsing the manifest and still get the results of the dbt job by inspecting the execution logs.
```

File: content/blogs/2024-04-02-dbt-kestra.md [hunk 9]
```diff
@@ -0,0 +1,228 @@
...
+      docker:
+        image: ghcr.io/kestra-io/dbt-duckdb:latest
+      commands:
+        - dbt deps
+        - dbt build
+```
+
+## Run a dbt project from a Github repository
+
+Here is an example using Kestra declarative syntax to run data ingestion pipelines with Airbyte and then run a dbt project from a GitHub repository
```
Comment (anna-geller) [inline content/blogs/2024-04-02-dbt-kestra.md:151]:
```suggestion
## Run a dbt project from a GitHub repository

Here is an example using Kestra declarative syntax to run data ingestion pipelines with Airbyte and then run a dbt project from a GitHub repository
```

File: content/blogs/2024-04-02-dbt-kestra.md [hunk 10]
```diff
@@ -0,0 +1,228 @@
...
+      - id: facebook-ads
+        type: io.kestra.plugin.airbyte.connections.Sync
+        connectionId: e3b1ce92-547c-436f-b1e8-23b6936c12ef
+
+  - id: dbt
+    type: io.kestra.core.tasks.flows.WorkingDirectory
+    tasks:
+    - id: cloneRepository
+      type: io.kestra.plugin.git.Clone
+      url: <https://github.com/kestra-io/dbt-demo>
```
Comment (anna-geller) [inline content/blogs/2024-04-02-dbt-kestra.md:178]:
```suggestion
      url: https://github.com/kestra-io/dbt-demo
```

File: content/blogs/2024-04-02-dbt-kestra.md [hunk 11]
```diff
@@ -0,0 +1,228 @@
...
+    values:
+      url: <http://host.docker.internal:8000/>
+      username: "{{ secret('AIRBYTE_USERNAME') }}"
+      password: "{{ secret('AIRBYTE_PASSWORD') }}"
+
+```
+
+Having the business logic managed by dbt and the orchestration logic in Kestra makes things way simpler to automate. Analytics engineers and data analysts can focus on the actual code that extracts business value from the data while data engineers can manage the orchestration layer and other projects altogether thanks to Kestra.
+
+By using a declarative language syntax, everyone can readily understand data pipelines. This simplifies collaboration, promotes knowledge sharing, and ultimately makes everyone more productive and confident in utilizing data within the company.
```
Comment (anna-geller) [inline content/blogs/2024-04-02-dbt-kestra.md:216]:
```suggestion
Having the business logic managed by dbt and the orchestration logic in Kestra makes things much simpler to automate. Thanks to Kestra, analytics engineers and data analysts can focus on the actual code that extracts business value from the data while data engineers can manage the orchestration layer and other projects.

By using a declarative language syntax, everyone can readily understand data pipelines. This simplifies collaboration, promotes knowledge sharing, and ultimately makes everyone more productive and confident in utilizing data within the company.
```

File: content/blogs/2024-04-02-dbt-kestra.md [hunk 12]
```diff
@@ -0,0 +1,228 @@
...
+
+Kestra simplifies the orchestration of dbt workflows, making it easier for users to manage and automate their data transformation processes. As we continue to witness the adoption and utilization of dbt within Kestra, we  forward to further improve our dbt plugin capabilities.
+
+If you want to learn more about the integrations between Kestra and dbt you can check our [library of dbt blueprints](https://kestra.io/blueprints?page=1&size=24&q=dbt).
+
+Check out the [plugin](https://kestra.io/plugins/plugin-dbt) documentation to get more informations about the dbt plugin.
+
+Join the Slack [community](https://kestra.io/slack) if you have any questions or need assistance.
+Follow us on [Twitter](https://twitter.com/kestra_io) for the latest news.
+Check the code in our [GitHub repository](https://github.com/kestra-io/kestra) and give us a star if you like the project.
```
Comment (anna-geller) [inline content/blogs/2024-04-02-dbt-kestra.md:228]:
```suggestion
Kestra simplifies the orchestration of dbt workflows, making it easier for users to manage and automate their data transformation processes. As we continue to witness the adoption and utilization of dbt within Kestra, we look forward to further improving our dbt plugin capabilities.

If you want to learn more about the integrations between Kestra and dbt, you can check our [library of dbt blueprints](https://kestra.io/blueprints?page=1&size=24&q=dbt).

Check out the [plugin](https://kestra.io/plugins/plugin-dbt) documentation for more information about the dbt plugin.

Join the Slack [community](https://kestra.io/slack) if you have any questions or need assistance.
Follow us on [Twitter](https://twitter.com/kestra_io) for the latest news.
Check the code in our [GitHub repository](https://github.com/kestra-io/kestra) and give us a star if you like the project.
```

**Review by anna-geller — APPROVED** [review]

File: content/blogs/2024-04-02-dbt-kestra.md [hunk 13]
```diff
@@ -0,0 +1,228 @@
+---
+title: "Orchestrate dbt Jobs with Kestra"
```
[the remark this answers is not in the document]
Comment (MartinRst) [reply content/blogs/2024-04-02-dbt-kestra.md:2]: Too long for the blog title! but I keep it for socials :)

**Pull request merged**